go/BUILD.bazel

```diff
@@ -0,0 +1,119 @@
+load("@bazel_skylib//rules:native_binary.bzl", "native_binary")
+load("@gazelle//:def.bzl", "gazelle")
+load("@rules_pkg//pkg:install.bzl", "pkg_install")
+load("@rules_pkg//pkg:mappings.bzl", "pkg_attributes", "pkg_filegroup", "pkg_files")
+load("@rules_pkg//pkg:zip.bzl", "pkg_zip")
+load("//:defs.bzl", "codeql_platform")
+
+gazelle(
+    name = "gazelle",
+    extra_args = ["go/extractor"],
+)
+
+_gen_binaries = [
+    "@rules_go//go",
+    ":gazelle",
+    "//go/extractor/cli/go-gen-dbscheme",
+]
+
+py_binary(
+    name = "gen",
+    srcs = ["gen.py"],
+    args = ["$(rlocationpath %s)" % bin for bin in _gen_binaries],
+    data = _gen_binaries,
+    deps = ["@rules_python//python/runfiles"],
+)
+
+# this is an internal copy of the dbscheme to be used by extractor-pack
```

Comment:
I re-read this comment 3 times, and it's not clear to me.
* Is the dbscheme that's checked into the repo generated by automation?
* If yes, we're still checking in the generated output, instead of piping through the generated file directly into our build?
* If yes, what's the motivation for separating these steps? That's a bit surprising, as I think in other cases, we don't check in generated data (and, while not always avoidable, I prefer not checking in generated files unless necessary as a default)

Reply:
This was a lesson learnt in the swift extractor.
* The dbscheme needs to be checked in, as `codeql`'s use as a checkout for QL libs requires that
* If the checked in dbscheme were to be used by bazel, in case of changes to it one would have to update that file prior to any `bazel` build step, otherwise `bazel` will see an outdated version. This can't (or rather shouldn't ever) be done by bazel itself, which means one would need to require a wrapper. In this case `make` would do the right thing, but stuff like `bazel build`/`bazel test` couldn't. This is a bummer for the development loop, especially when one is actively changing the dbscheme itself, because it can effectively mean two steps for each change (and you could forget to do one and miss an obvious problem).

For this reason we found it best to make bazel only depend on an internal generated instance of the dbscheme, and QL depend on the checked in one (with a PR check checking that everything is in order).

Reply:
Thanks for the explanation! I think the comment is better now, yes.